packages/edit-site/src/components/page-templates/index.js

```diff
@@ -170,20 +185,16 @@ export default function PageTemplates() {
 		[ postTypeActions, editAction ]
 	);
 
-	const onChangeView = useCallback(
```

Comment:
Why swap `useCallback` with `useEvent`?

Reply:
There's no need to generate a new callback instance if "view.type" changes or if "layout" changes or if "history" changes... We only want a stable reference that has access to all the latest props of the component. 

`useEvent` is great because it allows us to separate the "dependencies" of an effect from the actual code that needs to trigger as a result of that effect. We can control these two things separately leading to more clarity.

Reply:
Is it documented somewhere that this is how consumers should pass `onChangeView` - with stable reference?

If `DataView` cares about the `onChangeView` reference, maybe it's better if we handle it there. Just a random thought I had when viewing the code. We can discuss and handle it separately.

```js
const onChangeView = useEvent( onChangeViewProp );
```

Reply:
I see a number of components in DataViews using `memo` and using `onChangeView`, so memoization is needed. Now, it's true that we can do it internally but `useCallback` existed before this PR.